packages/tools/src/tools/segmentation/PaintFillTool.ts

```diff
@@ -77,18 +85,64 @@ class PaintFillTool extends BaseTool {
     }
 
     const { segmentationId, type } = activeSegmentationRepresentation;
+    if (type === SegmentationRepresentations.Contour) {
+      throw new Error('Not implemented yet');
+    }
+
     const segmentIndex =
       segmentIndexController.getActiveSegmentIndex(segmentationId);
     const segmentsLocked: number[] =
       segmentLocking.getLockedSegments(segmentationId);
     const { representationData } = getSegmentation(segmentationId);
 
-    const { volumeId } = representationData[type] as LabelmapSegmentationData;
-    const segmentation = cache.getVolume(volumeId);
-    const { dimensions, direction } = segmentation;
-    const scalarData = segmentation.getScalarData();
+    const labelmapData =
+      representationData[SegmentationRepresentations.Labelmap];
+
+    let dimensions, direction, scalarData, imageData;
+    const segmentationType = labelmapData.type;
+    const auxiliaryData = { type: '', imageIds: [] };
+    if (segmentationType === 'volume' || !segmentationType) {
+      const { volumeId } = representationData[
+        type
+      ] as LabelmapSegmentationDataVolume;
+      const segmentation = cache.getVolume(volumeId);
+      dimensions = segmentation.dimensions;
+      direction = segmentation.direction;
+      scalarData = segmentation.getScalarData();
+      imageData = segmentation.imageData;
+    } else {
+      const { referencedImageIds } =
+        labelmapData as LabelmapSegmentationDataStack;
+      const {
+        zSpacing,
+        sortedImageIds,
+        origin,
+        direction: calculatedDirection,
+      } = sortImageIds(referencedImageIds);
```

Comment:
I don't understand why it needs to sort referencedImageIds. It should just grab the current image id and gets its positionpatient, its direction and that is it. you can grab those from viewport.getImageData